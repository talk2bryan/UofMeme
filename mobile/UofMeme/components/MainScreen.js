import React, { Component } from "react";
import { View, Text, StyleSheet } from "react-native";
import { createBottomTabNavigator, createAppContainer } from "react-navigation";
import Ionicons from "react-native-vector-icons/Ionicons";

// import HomeTab from "./AppTabNavigator/HomeTab";
import HomeTabGrid from "./AppTabNavigator/HomeTabGrid";
import UploadTab from "./AppTabNavigator/UploadTab";

export default class MainScreen extends Component {
  render() {
    return <AppContainer />;
    // return <AppTabNavigator />;
  }
}

const AppTabNavigator = createBottomTabNavigator(
  {
    Home: HomeTabGrid,
    Upload: UploadTab
  },
  {
    defaultNavigationOptions: ({ navigation }) => ({
      tabBarIcon: ({ focused, horizontal, tintColor }) => {
        const { routeName } = navigation.state;
        let IconComponent = Ionicons;
        let iconName;
        if (routeName === "Home") {
          iconName = "ios-home";
          // Sometimes we want to add badges to some icons.
          // You can check the implementation below.
        } else if (routeName === "Upload") {
          iconName = "ios-cloud-upload";
        }

        // You can return any component that you like here!
        return <IconComponent name={iconName} size={25} color={tintColor} />;
      }
    }),
    animationEnabled: true,
    swipeEnabled: true,
    tabBarOptions: {
      activeTintColor: "orange",
      inactiveTintColor: "gray"
      // showLabel: false,
      // showIcon: true
    }
  }
);

const AppContainer = createAppContainer(AppTabNavigator);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center"
  }
});
