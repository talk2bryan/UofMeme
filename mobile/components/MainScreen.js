import React, { Component } from "react";
import { View, Text, StyleSheet, Platform } from "react-native";

import { Icon } from "native-base";
import { TabNavigator } from "react-navigation";

import HomeTab from "./AppTabNavigator/HomeTab";
import UploadTab from "./AppTabNavigator/UploadTab";
import CreateTab from "./AppTabNavigator/CreateTab";

class MainScreen extends Component {
  static navigationOptions = {
    title: "U of Meme",
    headerRight: <Icon style={{ paddingRight: 15 }} name="ios-camera-outline" />
  };

  render() {
    return <AppTabNavigator />;
  }
}

//make this component available to the app
export default MainScreen;

const AppTabNavigator = TabNavigator(
  {
    HomeTab: {
      screen: HomeTab
    },
    UploadTab: {
      screen: UploadTab
    },
    CreateTab: {
      screen: CreateTab
    }
  },
  {
    tabBarPosition: "bottom",
    animationEnabled: true,
    swipeEnabled: true,
    tabBarOptions: {
      activeTintColor: "#000",
      inactiveTintColor: "#7d7b7b",
      showLabel: false,
      showIcon: true,
      labelStyle: {
        fontSize: 12
      },
      style: {
        ...Platform.select({
          android: {
            backgroundColor: "#fff"
          }
        })
      }
    }
  }
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  }
});
