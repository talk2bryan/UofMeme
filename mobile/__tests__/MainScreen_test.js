import React from 'react';
import "isomorphic-fetch";
import renderer from 'react-test-renderer';

import MainScreen from '../components/MainScreen';

test('renders correctly', () => {
	const tree = renderer.create(<MainScreen/>).toJSON();
	expect(tree).toMatchSnapshot();
});

