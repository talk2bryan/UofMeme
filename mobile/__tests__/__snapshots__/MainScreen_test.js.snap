// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  loaded={
    Array [
      0,
    ]
  }
  onLayout={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      Object {
        "flex": 1,
      },
    ]
  }
>
  <RCTScrollView
    alwaysBounceHorizontal={false}
    automaticallyAdjustContentInsets={false}
    bounces={false}
    contentContainerStyle={
      Object {
        "flex": 1,
      }
    }
    contentOffset={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    directionalLockEnabled={true}
    horizontal={true}
    keyboardDismissMode="on-drag"
    keyboardShouldPersistTaps="always"
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    overScrollMode="never"
    pagingEnabled={true}
    scrollEnabled={true}
    scrollEventThrottle={1}
    scrollsToTop={false}
    showsHorizontalScrollIndicator={false}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "overflow": "hidden",
          }
        }
        testID={undefined}
      >
        <View
          collapsable={false}
          removeClippedSubviews={false}
          style={
            Object {
              "flex": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "height": 1334,
                  },
                  Object {
                    "backgroundColor": "white",
                    "flex": 1,
                  },
                ]
              }
              virtual={undefined}
            >
              <RCTScrollView
                automaticallyAdjustContentInsets={false}
                contentContainerStyle={undefined}
                contentInset={
                  Object {
                    "bottom": 0,
                  }
                }
                enableAutoAutomaticScroll={true}
                enableResetScrollToCoords={true}
                extraHeight={75}
                extraScrollHeight={0}
                getScrollResponder={[Function]}
                handleOnScroll={[Function]}
                keyboardDismissMode="interactive"
                keyboardOpeningTime={250}
                keyboardShouldPersistTaps="handled"
                keyboardSpace={0}
                onScroll={[Function]}
                resetKeyboardSpace={[Function]}
                resetScrollToCoords={
                  Object {
                    "x": 0,
                    "y": 0,
                  }
                }
                scrollEventThrottle={1}
                scrollForExtraHeightOnAndroid={[Function]}
                scrollToEnd={[Function]}
                scrollToFocusedInput={[Function]}
                scrollToPosition={[Function]}
                showsVerticalScrollIndicator={true}
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "flex": 1,
                  }
                }
                viewIsInsideTabBar={false}
                virtual={undefined}
              >
                <View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "paddingTop": 20,
                      }
                    }
                  >
                    <ActivityIndicator
                      animating={true}
                      color="#999999"
                      hidesWhenStopped={true}
                      size="small"
                    />
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </View>
        </View>
      </View>
      <View
        style={null}
        testID={undefined}
      />
      <View
        style={null}
        testID={undefined}
      />
    </View>
  </RCTScrollView>
  <View
    collapsable={undefined}
    style={undefined}
  >
    <View
      collapsable={undefined}
      onLayout={[Function]}
      pointerEvents="box-none"
      style={
        Object {
          "backgroundColor": "#F7F7F7",
          "borderTopColor": "rgba(0, 0, 0, .3)",
          "borderTopWidth": 0.5,
          "flexDirection": "row",
          "height": 49,
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 29,
                },
                Object {
                  "flex": 1,
                },
                false,
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 29,
                },
                Object {
                  "flex": 1,
                },
                false,
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(0, 0, 0, 0)",
            "flex": 1,
          }
        }
        testID={undefined}
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flex": 1,
              },
              Object {
                "flexDirection": "column",
                "justifyContent": "flex-end",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "height": 29,
                },
                Object {
                  "flex": 1,
                },
                false,
              ]
            }
          >
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 0,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
            <View
              collapsable={undefined}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "center",
                  "height": "100%",
                  "justifyContent": "center",
                  "opacity": 1,
                  "position": "absolute",
                  "width": "100%",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
